state-chain/pallets/cf-environment/src/lib.rs

```diff
@@ -245,6 +274,47 @@ pub mod pallet {
 	/// Contains the network environment for this runtime.
 	pub type ChainflipNetworkEnvironment<T> = StorageValue<_, NetworkEnvironment, ValueQuery>;
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_idle(block_number: BlockNumberFor<T>, _remaining_weight: Weight) -> Weight {
```

Comment:
yeah, I left the benchmarking for later. I'll open an issue for it.

Reply:
the benchmarking ticket could have been assigned to me and I could've taken it up immediately after this (and the election ticket) but sure I can do it as part of this PR if you insist.

I will come back to it after working on election ticket. Since that is closely linked to the functionality in this PR, some of the functionality in this PR might change and so I would do the benchmarks once the functionality is final.